Unify DAG creation/database cleaning fixtures for testing

## Description

<!-- Describe the feature and how it solves the problem. -->

We have a few places in our catalog tests with very similar logic for creating the DAG necessary for running some tests. Examples:

https://github.com/WordPress/openverse/blob/9056029e6c354fe2dbf5b0ad49d60648bd380d27/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py#L23-L34

https://github.com/WordPress/openverse/blob/9056029e6c354fe2dbf5b0ad49d60648bd380d27/catalog/tests/dags/data_refresh/test_dag_factory.py#L22-L25

https://github.com/WordPress/openverse/blob/9056029e6c354fe2dbf5b0ad49d60648bd380d27/catalog/tests/dags/common/test_ingestion_server.py#L21-L31

https://github.com/WordPress/openverse/blob/9056029e6c354fe2dbf5b0ad49d60648bd380d27/catalog/tests/dags/common/sensors/test_utils.py#L18-L21

https://github.com/WordPress/openverse/blob/9056029e6c354fe2dbf5b0ad49d60648bd380d27/catalog/tests/dags/providers/test_provider_dag_factory.py#L27-L37


Although there are some slight differences, it would be ideal if we could find a way to combine these into a single fixture that's used within all the affected tests, rather than redefining the fixture for every test.

## Additional context

<!-- Add any other context about the feature here; or delete the section entirely. -->

Came up in discussing https://github.com/WordPress/openverse/pull/2209#discussion_r1232622950

<!-- If you would like to work on this, please comment below separately. -->

```
    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) can't adapt type 'function'
E       [SQL: DELETE FROM dag_run WHERE dag_run.dag_id = %(dag_id_1)s]
E       [parameters: {'dag_id_1': <function test_dag_id at 0x7f94651632e0>}]
E       (Background on this error at: https://sqlalche.me/e/14/f405)

/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py:736: ProgrammingError

```
I tried using the `request` fixture and there is an error when calling `getattr`. I have created a pull request using an override method.([link](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#override-a-fixture-on-a-test-module-level)).

```
    def do_execute(self, cursor, statement, parameters, context=None):
>       cursor.execute(statement, parameters)
E       sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) can't adapt type 'function'
E       [SQL: DELETE FROM dag_run WHERE dag_run.dag_id = %(dag_id_1)s]
E       [parameters: {'dag_id_1': <function test_dag_id at 0x7f94651632e0>}]
E       (Background on this error at: https://sqlalche.me/e/14/f405)

/home/airflow/.local/lib/python3.10/site-packages/sqlalchemy/engine/default.py:736: ProgrammingError

```
I tried using the `request` fixture and there is an error when calling `getattr`. I have created a pull request using an override method.([link](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#override-a-fixture-on-a-test-module-level))

--- a/catalog/tests/conftest.py
+++ b/catalog/tests/conftest.py
@@ -1,4 +1,6 @@
 import pytest
+from airflow.utils.session import create_session
+from airflow.models import DagRun, Pool, TaskInstance
 
 
 def pytest_addoption(parser):
@@ -24,3 +26,31 @@ def pytest_addoption(parser):
 # Use this decorator on tests which are expected to take a long time and would best be
 # run on CI only
 mark_extended = pytest.mark.skipif("not config.getoption('extended')")
+
+@pytest.fixture()
+def get_test_dag_id():
+    return ""
+
+@pytest.fixture()
+def get_test_pool():
+    return ""
+@pytest.fixture()
+def isTaskInstance():
+    return False
+
+@pytest.fixture()
+def isPool():
+    return False
+
+@pytest.fixture(autouse=True)
+def clean_db(get_test_dag_id, get_test_pool, isTaskInstance, isPool):
+    with create_session() as session:
+        # synchronize_session='fetch' required here to refresh models
+        # https://stackoverflow.com/a/51222378 CC BY-SA 4.0
+        session.query(DagRun).filter(DagRun.dag_id == get_test_dag_id).delete()
+        if isTaskInstance:
+            session.query(TaskInstance).filter(
+                TaskInstance.dag_id == get_test_dag_id
+            ).delete(synchronize_session="fetch")
+        if isPool:
+            session.query(Pool).filter(id == get_test_pool).delete()

--- a/catalog/tests/conftest.py
+++ b/catalog/tests/conftest.py
@@ -36,26 +36,26 @@ def _normalize_test_module_name(request) -> str:
 
 
 @pytest.fixture
-def get_test_dag_id(request):
+def sample_dag_id_fixture(request):
     return f"{_normalize_test_module_name(request)}_dag"
 
 
 @pytest.fixture
-def get_test_pool(request):
+def sample_pool_fixture(request):
     return f"{_normalize_test_module_name(request)}_pool"
 
 
 @pytest.fixture
-def clean_db(get_test_dag_id, get_test_pool):
+def clean_db(sample_dag_id_fixture, sample_pool_fixture):
     with create_session() as session:
         # synchronize_session='fetch' required here to refresh models
         # https://stackoverflow.com/a/51222378 CC BY-SA 4.0
-        session.query(DagRun).filter(DagRun.dag_id.startswith(get_test_dag_id)).delete(
-            synchronize_session="fetch"
-        )
-        session.query(TaskInstance).filter(
-            TaskInstance.dag_id.startswith(get_test_dag_id)
+        session.query(DagRun).filter(
+            DagRun.dag_id.startswith(sample_dag_id_fixture)
         ).delete(synchronize_session="fetch")
-        session.query(Pool).filter(Pool.pool.startswith(get_test_pool)).delete(
+        session.query(TaskInstance).filter(
+            TaskInstance.dag_id.startswith(sample_dag_id_fixture)
+        ).delete(synchronize_session="fetch")
+        session.query(Pool).filter(Pool.pool.startswith(sample_pool_fixture)).delete(
             synchronize_session="fetch"
         )

--- a/catalog/tests/conftest.py
+++ b/catalog/tests/conftest.py
@@ -28,35 +28,34 @@ def pytest_addoption(parser):
 mark_extended = pytest.mark.skipif("not config.getoption('extended')")
 
 
-@pytest.fixture()
-def get_test_dag_id():
-    return ""
+def _normalize_test_module_name(request) -> str:
+    # Extract the test name
+    name = request.module.__name__
+    # Replace periods with two underscores
+    return name.replace(".", "__")
 
 
-@pytest.fixture()
-def get_test_pool():
-    return ""
+@pytest.fixture
+def get_test_dag_id(request):
+    return f"{_normalize_test_module_name(request)}_dag"
 
 
-@pytest.fixture()
-def isTaskInstance():
-    return False
+@pytest.fixture
+def get_test_pool(request):
+    return f"{_normalize_test_module_name(request)}_pool"
 
 
-@pytest.fixture()
-def isPool():
-    return False
-
-
-@pytest.fixture(autouse=True)
-def clean_db(get_test_dag_id, get_test_pool, isTaskInstance, isPool):
+@pytest.fixture
+def clean_db(get_test_dag_id, get_test_pool):
     with create_session() as session:
         # synchronize_session='fetch' required here to refresh models
         # https://stackoverflow.com/a/51222378 CC BY-SA 4.0
-        session.query(DagRun).filter(DagRun.dag_id == get_test_dag_id).delete()
-        if isTaskInstance:
-            session.query(TaskInstance).filter(
-                TaskInstance.dag_id == get_test_dag_id
-            ).delete(synchronize_session="fetch")
-        if isPool:
-            session.query(Pool).filter(id == get_test_pool).delete()
+        session.query(DagRun).filter(DagRun.dag_id.startswith(get_test_dag_id)).delete(
+            synchronize_session="fetch"
+        )
+        session.query(TaskInstance).filter(
+            TaskInstance.dag_id.startswith(get_test_dag_id)
+        ).delete(synchronize_session="fetch")
+        session.query(Pool).filter(Pool.pool.startswith(get_test_pool)).delete(
+            synchronize_session="fetch"
+        )

--- a/catalog/tests/conftest.py
+++ b/catalog/tests/conftest.py
@@ -1,6 +1,6 @@
 import pytest
-from airflow.utils.session import create_session
 from airflow.models import DagRun, Pool, TaskInstance
+from airflow.utils.session import create_session
 
 
 def pytest_addoption(parser):
@@ -27,20 +27,26 @@ def pytest_addoption(parser):
 # run on CI only
 mark_extended = pytest.mark.skipif("not config.getoption('extended')")
 
+
 @pytest.fixture()
 def get_test_dag_id():
     return ""
 
+
 @pytest.fixture()
 def get_test_pool():
     return ""
+
+
 @pytest.fixture()
 def isTaskInstance():
     return False
 
+
 @pytest.fixture()
 def isPool():
     return False
+
 
 @pytest.fixture(autouse=True)
 def clean_db(get_test_dag_id, get_test_pool, isTaskInstance, isPool):

--- a/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
+++ b/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
@@ -14,29 +14,11 @@ from common.sensors.single_run_external_dags_sensor import SingleRunExternalDAGs
 
 DEFAULT_DATE = datetime(2022, 1, 1)
 TEST_TASK_ID = "wait_task"
-TEST_POOL = "single_run_external_dags_sensor_test_pool"
+TEST_POOL = (
+    "catalog__tests__dags__common__sensors__test_single_run_external_dags_sensor_pool"
+)
 DEV_NULL = "/dev/null"
-DAG_PREFIX = "sreds"  # single_run_external_dags_sensor
-
-
-@pytest.fixture()
-def get_test_pool():
-    return TEST_POOL
-
-
-@pytest.fixture()
-def get_test_dag_id():
-    return DAG_PREFIX
-
-
-@pytest.fixture()
-def isTaskInstance():
-    return False
-
-
-@pytest.fixture()
-def isPool():
-    return True
+DAG_PREFIX = "catalog__tests__dags__common__sensors__test_single_run_external_dags_sensor_dag"  # single_run_external_dags_sensor
 
 
 def run_sensor(sensor):
@@ -80,6 +62,7 @@ def create_dagrun(dag, dag_state):
     )
 
 
+@pytest.mark.usefixtures("clean_db")
 # This appears to be coming from Airflow internals during testing as a result of
 # loading the example DAGs:
 # /opt/airflow/.local/lib/python3.10/site-packages/airflow/example_dags/example_subdag_operator.py:43: RemovedInAirflow3Warning  # noqa: E501

--- a/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
+++ b/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
@@ -18,21 +18,26 @@ TEST_POOL = "single_run_external_dags_sensor_test_pool"
 DEV_NULL = "/dev/null"
 DAG_PREFIX = "sreds"  # single_run_external_dags_sensor
 
+
 @pytest.fixture()
 def get_test_pool():
     return TEST_POOL
+
 
 @pytest.fixture()
 def get_test_dag_id():
     return DAG_PREFIX
 
+
 @pytest.fixture()
 def isTaskInstance():
     return False
 
+
 @pytest.fixture()
 def isPool():
     return True
+
 
 def run_sensor(sensor):
     sensor.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)

--- a/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
+++ b/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
@@ -14,10 +14,13 @@ from common.sensors.single_run_external_dags_sensor import SingleRunExternalDAGs
 
 DEFAULT_DATE = datetime(2022, 1, 1)
 TEST_TASK_ID = "wait_task"
+DEV_NULL = "/dev/null"
+
+# unittest.TestCase only allow auto-use fixture which can't retrieve the declared fixtures on conftest.py
+# TODO: TEST_POOL/DAG_PREFIX constants can be remove after unittest.TestCase are converted to pytest.
 TEST_POOL = (
     "catalog__tests__dags__common__sensors__test_single_run_external_dags_sensor_pool"
 )
-DEV_NULL = "/dev/null"
 DAG_PREFIX = "catalog__tests__dags__common__sensors__test_single_run_external_dags_sensor_dag"  # single_run_external_dags_sensor
 
 

--- a/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
+++ b/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
@@ -3,9 +3,8 @@ import unittest
 
 import pytest
 from airflow.exceptions import AirflowException
-from airflow.models import DagBag, DagRun, Pool, TaskInstance
+from airflow.models import DagBag, Pool
 from airflow.models.dag import DAG
-from airflow.utils.session import create_session
 from airflow.utils.state import State
 from airflow.utils.timezone import datetime
 from airflow.utils.types import DagRunType

--- a/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
+++ b/catalog/tests/dags/common/sensors/test_single_run_external_dags_sensor.py
@@ -19,20 +19,21 @@ TEST_POOL = "single_run_external_dags_sensor_test_pool"
 DEV_NULL = "/dev/null"
 DAG_PREFIX = "sreds"  # single_run_external_dags_sensor
 
+@pytest.fixture()
+def get_test_pool():
+    return TEST_POOL
 
-@pytest.fixture(autouse=True)
-def clean_db():
-    with create_session() as session:
-        # synchronize_session='fetch' required here to refresh models
-        # https://stackoverflow.com/a/51222378 CC BY-SA 4.0
-        session.query(DagRun).filter(DagRun.dag_id.startswith(DAG_PREFIX)).delete(
-            synchronize_session="fetch"
-        )
-        session.query(TaskInstance).filter(
-            TaskInstance.dag_id.startswith(DAG_PREFIX)
-        ).delete(synchronize_session="fetch")
-        session.query(Pool).filter(id == TEST_POOL).delete()
+@pytest.fixture()
+def get_test_dag_id():
+    return DAG_PREFIX
 
+@pytest.fixture()
+def isTaskInstance():
+    return False
+
+@pytest.fixture()
+def isPool():
+    return True
 
 def run_sensor(sensor):
     sensor.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)

--- a/catalog/tests/dags/common/sensors/test_utils.py
+++ b/catalog/tests/dags/common/sensors/test_utils.py
@@ -1,6 +1,5 @@
 from datetime import timedelta
 
-import pytest
 from airflow.models.dag import DAG
 from airflow.utils.state import State
 from airflow.utils.timezone import datetime
@@ -13,13 +12,8 @@ TEST_DAG_ID = "data_refresh_dag_factory_test_dag"
 TEST_DAG = DAG(TEST_DAG_ID, default_args={"owner": "airflow"})
 
 
-@pytest.fixture()
-def get_test_dag_id():
-    return TEST_DAG_ID
-
-
-def _create_dagrun(start_date, conf={}):
-    return TEST_DAG.create_dagrun(
+def _create_dagrun(start_date, get_test_dag_id, conf={}):
+    return DAG(get_test_dag_id, default_args={"owner": "airflow"}).create_dagrun(
         start_date=start_date,
         execution_date=start_date,
         data_interval=(start_date, start_date),
@@ -29,14 +23,17 @@ def _create_dagrun(start_date, conf={}):
     )
 
 
-def test_get_most_recent_dag_run_returns_most_recent_execution_date():
+def test_get_most_recent_dag_run_returns_most_recent_execution_date(
+    get_test_dag_id, clean_db
+):
     most_recent = datetime(2023, 5, 10)
     for i in range(3):
-        _create_dagrun(most_recent - timedelta(days=i))
+        _create_dagrun(most_recent - timedelta(days=i), get_test_dag_id)
+    assert get_most_recent_dag_run(get_test_dag_id) == most_recent
 
-    assert get_most_recent_dag_run(TEST_DAG_ID) == most_recent
 
-
-def test_get_most_recent_dag_run_returns_empty_list_when_no_runs():
+def test_get_most_recent_dag_run_returns_empty_list_when_no_runs(
+    get_test_dag_id, clean_db
+):
     # Relies on ``clean_db`` cleaning up DagRuns from other tests
-    assert get_most_recent_dag_run(TEST_DAG_ID) == []
+    assert get_most_recent_dag_run(get_test_dag_id) == []

--- a/catalog/tests/dags/common/sensors/test_utils.py
+++ b/catalog/tests/dags/common/sensors/test_utils.py
@@ -12,8 +12,8 @@ TEST_DAG_ID = "data_refresh_dag_factory_test_dag"
 TEST_DAG = DAG(TEST_DAG_ID, default_args={"owner": "airflow"})
 
 
-def _create_dagrun(start_date, get_test_dag_id, conf={}):
-    return DAG(get_test_dag_id, default_args={"owner": "airflow"}).create_dagrun(
+def _create_dagrun(start_date, sample_dag_id_fixture, conf={}):
+    return DAG(sample_dag_id_fixture, default_args={"owner": "airflow"}).create_dagrun(
         start_date=start_date,
         execution_date=start_date,
         data_interval=(start_date, start_date),
@@ -24,16 +24,16 @@ def _create_dagrun(start_date, get_test_dag_id, conf={}):
 
 
 def test_get_most_recent_dag_run_returns_most_recent_execution_date(
-    get_test_dag_id, clean_db
+    sample_dag_id_fixture, clean_db
 ):
     most_recent = datetime(2023, 5, 10)
     for i in range(3):
-        _create_dagrun(most_recent - timedelta(days=i), get_test_dag_id)
-    assert get_most_recent_dag_run(get_test_dag_id) == most_recent
+        _create_dagrun(most_recent - timedelta(days=i), sample_dag_id_fixture)
+    assert get_most_recent_dag_run(sample_dag_id_fixture) == most_recent
 
 
 def test_get_most_recent_dag_run_returns_empty_list_when_no_runs(
-    get_test_dag_id, clean_db
+    sample_dag_id_fixture, clean_db
 ):
     # Relies on ``clean_db`` cleaning up DagRuns from other tests
-    assert get_most_recent_dag_run(get_test_dag_id) == []
+    assert get_most_recent_dag_run(sample_dag_id_fixture) == []

--- a/catalog/tests/dags/common/sensors/test_utils.py
+++ b/catalog/tests/dags/common/sensors/test_utils.py
@@ -14,12 +14,9 @@ from common.sensors.utils import get_most_recent_dag_run
 TEST_DAG_ID = "data_refresh_dag_factory_test_dag"
 TEST_DAG = DAG(TEST_DAG_ID, default_args={"owner": "airflow"})
 
-
-@pytest.fixture(autouse=True)
-def clean_db():
-    with create_session() as session:
-        session.query(DagRun).filter(DagRun.dag_id == TEST_DAG_ID).delete()
-
+@pytest.fixture()
+def get_test_dag_id():
+    return TEST_DAG_ID
 
 def _create_dagrun(start_date, conf={}):
     return TEST_DAG.create_dagrun(

--- a/catalog/tests/dags/common/sensors/test_utils.py
+++ b/catalog/tests/dags/common/sensors/test_utils.py
@@ -1,9 +1,7 @@
 from datetime import timedelta
 
 import pytest
-from airflow.models import DagRun
 from airflow.models.dag import DAG
-from airflow.utils.session import create_session
 from airflow.utils.state import State
 from airflow.utils.timezone import datetime
 from airflow.utils.types import DagRunType

--- a/catalog/tests/dags/common/sensors/test_utils.py
+++ b/catalog/tests/dags/common/sensors/test_utils.py
@@ -12,9 +12,11 @@ from common.sensors.utils import get_most_recent_dag_run
 TEST_DAG_ID = "data_refresh_dag_factory_test_dag"
 TEST_DAG = DAG(TEST_DAG_ID, default_args={"owner": "airflow"})
 
+
 @pytest.fixture()
 def get_test_dag_id():
     return TEST_DAG_ID
+
 
 def _create_dagrun(start_date, conf={}):
     return TEST_DAG.create_dagrun(

--- a/catalog/tests/dags/common/test_ingestion_server.py
+++ b/catalog/tests/dags/common/test_ingestion_server.py
@@ -16,9 +16,11 @@ TEST_START_DATE = datetime(2022, 2, 1, 0, 0, 0)
 
 
 @pytest.fixture()
-def index_readiness_dag(get_test_dag_id, clean_db):
+def index_readiness_dag(sample_dag_id_fixture, clean_db):
     # Create a DAG that just has an index_readiness_check task
-    with DAG(dag_id=get_test_dag_id, schedule=None, start_date=TEST_START_DATE) as dag:
+    with DAG(
+        dag_id=sample_dag_id_fixture, schedule=None, start_date=TEST_START_DATE
+    ) as dag:
         ingestion_server.index_readiness_check(
             media_type="image", index_suffix="my_test_suffix", timeout=timedelta(days=1)
         )

--- a/catalog/tests/dags/common/test_ingestion_server.py
+++ b/catalog/tests/dags/common/test_ingestion_server.py
@@ -18,18 +18,13 @@ TEST_START_DATE = datetime(2022, 2, 1, 0, 0, 0)
 TEST_DAG_ID = "api_healthcheck_test_dag"
 
 
-@pytest.fixture(autouse=True)
-def clean_db():
-    with create_session() as session:
-        # synchronize_session='fetch' required here to refresh models
-        # https://stackoverflow.com/a/51222378 CC BY-SA 4.0
-        session.query(DagRun).filter(DagRun.dag_id.startswith(TEST_DAG_ID)).delete(
-            synchronize_session="fetch"
-        )
-        session.query(TaskInstance).filter(
-            TaskInstance.dag_id.startswith(TEST_DAG_ID)
-        ).delete(synchronize_session="fetch")
+@pytest.fixture()
+def get_test_dag_id():
+    return TEST_DAG_ID
 
+@pytest.fixture()
+def isTaskInstance():
+    return True
 
 @pytest.fixture()
 def index_readiness_dag():

--- a/catalog/tests/dags/common/test_ingestion_server.py
+++ b/catalog/tests/dags/common/test_ingestion_server.py
@@ -5,7 +5,6 @@ import pytest
 import requests
 from airflow.exceptions import AirflowSkipException
 from airflow.models.dag import DAG
-from airflow.utils.session import create_session
 from airflow.utils.state import DagRunState, TaskInstanceState
 from airflow.utils.timezone import datetime
 from airflow.utils.types import DagRunType
@@ -21,9 +20,11 @@ TEST_DAG_ID = "api_healthcheck_test_dag"
 def get_test_dag_id():
     return TEST_DAG_ID
 
+
 @pytest.fixture()
 def isTaskInstance():
     return True
+
 
 @pytest.fixture()
 def index_readiness_dag():

--- a/catalog/tests/dags/common/test_ingestion_server.py
+++ b/catalog/tests/dags/common/test_ingestion_server.py
@@ -4,7 +4,6 @@ from unittest import mock
 import pytest
 import requests
 from airflow.exceptions import AirflowSkipException
-from airflow.models import DagRun, TaskInstance
 from airflow.models.dag import DAG
 from airflow.utils.session import create_session
 from airflow.utils.state import DagRunState, TaskInstanceState

--- a/catalog/tests/dags/common/test_ingestion_server.py
+++ b/catalog/tests/dags/common/test_ingestion_server.py
@@ -13,27 +13,15 @@ from common import ingestion_server
 
 
 TEST_START_DATE = datetime(2022, 2, 1, 0, 0, 0)
-TEST_DAG_ID = "api_healthcheck_test_dag"
 
 
 @pytest.fixture()
-def get_test_dag_id():
-    return TEST_DAG_ID
-
-
-@pytest.fixture()
-def isTaskInstance():
-    return True
-
-
-@pytest.fixture()
-def index_readiness_dag():
+def index_readiness_dag(get_test_dag_id, clean_db):
     # Create a DAG that just has an index_readiness_check task
-    with DAG(dag_id=TEST_DAG_ID, schedule=None, start_date=TEST_START_DATE) as dag:
+    with DAG(dag_id=get_test_dag_id, schedule=None, start_date=TEST_START_DATE) as dag:
         ingestion_server.index_readiness_check(
             media_type="image", index_suffix="my_test_suffix", timeout=timedelta(days=1)
         )
-
     return dag
 
 

--- a/catalog/tests/dags/providers/test_provider_dag_factory.py
+++ b/catalog/tests/dags/providers/test_provider_dag_factory.py
@@ -21,13 +21,16 @@ from providers.provider_workflows import ProviderWorkflow
 
 TEST_DAG_ID = "test_provider_dag_factory"
 
+
 @pytest.fixture()
 def get_test_dag_id():
     return TEST_DAG_ID
 
+
 @pytest.fixture()
 def isTaskInstance():
     return True
+
 
 @mark_extended
 @pytest.mark.parametrize(

--- a/catalog/tests/dags/providers/test_provider_dag_factory.py
+++ b/catalog/tests/dags/providers/test_provider_dag_factory.py
@@ -5,9 +5,7 @@ import pytest
 from airflow import DAG
 from airflow.exceptions import AirflowSkipException, BackfillUnfinished
 from airflow.executors.debug_executor import DebugExecutor
-from airflow.models import DagRun, TaskInstance
 from airflow.operators.empty import EmptyOperator
-from airflow.utils.session import create_session
 from pendulum import now
 
 from catalog.tests.conftest import mark_extended

--- a/catalog/tests/dags/providers/test_provider_dag_factory.py
+++ b/catalog/tests/dags/providers/test_provider_dag_factory.py
@@ -19,19 +19,6 @@ from providers.provider_reingestion_workflows import ProviderReingestionWorkflow
 from providers.provider_workflows import ProviderWorkflow
 
 
-TEST_DAG_ID = "test_provider_dag_factory"
-
-
-@pytest.fixture()
-def get_test_dag_id():
-    return TEST_DAG_ID
-
-
-@pytest.fixture()
-def isTaskInstance():
-    return True
-
-
 @mark_extended
 @pytest.mark.parametrize(
     "side_effect",

--- a/catalog/tests/dags/providers/test_provider_dag_factory.py
+++ b/catalog/tests/dags/providers/test_provider_dag_factory.py
@@ -21,21 +21,15 @@ from providers.provider_reingestion_workflows import ProviderReingestionWorkflow
 from providers.provider_workflows import ProviderWorkflow
 
 
-DAG_ID = "test_provider_dag_factory"
-
-
-def _clean_dag_from_db():
-    with create_session() as session:
-        session.query(DagRun).filter(DagRun.dag_id == DAG_ID).delete()
-        session.query(TaskInstance).filter(TaskInstance.dag_id == DAG_ID).delete()
-
+TEST_DAG_ID = "test_provider_dag_factory"
 
 @pytest.fixture()
-def clean_db():
-    _clean_dag_from_db()
-    yield
-    _clean_dag_from_db()
+def get_test_dag_id():
+    return TEST_DAG_ID
 
+@pytest.fixture()
+def isTaskInstance():
+    return True
 
 @mark_extended
 @pytest.mark.parametrize(

